File: src/main/webapp/WEB-INF/static/core/service/JogadorInfoEdicaoService.js
/**
 * JogadorInfoEdicaoService
 */
app.factory('JogadorInfoEdicaoService', ['$http', function($http) {

	var _serverURL = function(url) {
		return SERVER_APP + '/jogadorinfoedicao' + url;
	};
	
	var _getByEdicao = function(edicao) {
		return $http.get(_serverURL('/get/by/edicao/'+edicao.id));
	};
	
	var _getByEdicaoArtilharia = function(edicao) {
		return $http.get(_serverURL('/get/by/edicao/'+edicao.id+'/artilharia'));
	};
	
	var _getByEdicaoAndJogador = function(edicao, jogador) {
		return $http.get(_serverURL('/get/by/edicao/'+edicao.id+'/jogador/'+jogador.id));
	};
	
	return {
		
		getByEdicao : _getByEdicao,
		
		getByEdicaoArtilharia : _getByEdicaoArtilharia,
		
		getByEdicaoAndJogador : _getByEdicaoAndJogador
		
	};
	
}]);

File: src/main/java/com/diegolirio/jcampeonato/service/JogadorInfoEdicaoService.java
package com.diegolirio.jcampeonato.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.diegolirio.jcampeonato.dao.JogadorInfoEdicaoDao;
import com.diegolirio.jcampeonato.model.Edicao;
import com.diegolirio.jcampeonato.model.Jogador;
import com.diegolirio.jcampeonato.model.JogadorInfoEdicao;

@Service("jogadorInfoEdicaoService")
public class JogadorInfoEdicaoService extends AbstractGenericService<JogadorInfoEdicao>{

	@Autowired
	private JogadorInfoEdicaoDao jogadorInfoEdicaoDao;

	public List<JogadorInfoEdicao> getByEdicao(Edicao edicao) {
		return this.jogadorInfoEdicaoDao.getByEdicao(edicao);
	}

	public JogadorInfoEdicao getByEdicaoAndJogador(Edicao edicao, Jogador jogador) {
		return this.jogadorInfoEdicaoDao.getByEdicaoAndJogador(edicao, jogador);
	}

}
